pkg/validation/validation.go

```diff
@@ -71,6 +74,8 @@ func ValidateLabelSet(labels map[string]string) (warnings []string, err error) {
 	return warnings, err
 }
 
+// ValidateLabel will check if the the label key given has been deprecated,
+// returning the appropriate deprecation notice if so.
 func ValidateLabel(k, v string) (warnings []string, err error) {
```

Comment:
Could we rename this? I gather from the `TODOs` it is supposed to perform generic validation, but if it only checks for deprecation it might be more appropriate to name it `IsDeprecatedLabel()` or something so the call site is a little more truthful!

Feel free to open a ticket for this though - it's properly out of scope.

Reply:
I had thought about opening up a ticket to include validation on the values, but I think a rename makes sense. Alternatively we could probably trust in the server-side validation here and in doing so avoid maintaining deprecation checks/validation in multiple places.

Reply:
Yeah if the API does validation we shouldn't duplicate it here - are deprecation warnings client-side only though?

Reply:
Seems like deprecation warnings are client-side only, will leave rename and leave this in for now.